### ISSUE TITLE
Junit XML reporter

### ISSUE THREAD

**Issue by krvajalmiguelangel** [krvajalmiguelangel/Catch#22]
<a href="https://github.com/philsquared"><img src="https://avatars.githubusercontent.com/u/92878?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [philsquared](https://github.com/philsquared)**
_Friday Nov 19, 2010 at 08:31 GMT_
_Originally opened as https://github.com/philsquared/Catch/issues/5_

----

It should be possible to obtain xml reports that follow the schema for JUnit - for consumption by third party tools

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/philsquared"><img src="https://avatars.githubusercontent.com/u/92878?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [philsquared](https://github.com/philsquared)**
_Wednesday Dec 15, 2010 at 19:29 GMT_

----

First cut of this has been committed, but it's not fully tested yet and more examples of real JUnit output would be useful

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/wichert"><img src="https://avatars.githubusercontent.com/u/199657?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [wichert](https://github.com/wichert)**
_Sunday Apr 24, 2011 at 19:19 GMT_

----

Jenkins does not like the current output:

```
ERROR: Publisher hudson.tasks.junit.JUnitResultArchiver aborted due to exception
java.lang.NullPointerException
    at hudson.tasks.junit.CaseResult.getPackageName(CaseResult.java:266)
    at hudson.tasks.junit.TestResult.tally(TestResult.java:500)
    at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:115)
    at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:87)
    at hudson.FilePath.act(FilePath.java:757)
    at hudson.FilePath.act(FilePath.java:739)
    at hudson.tasks.junit.JUnitParser.parse(JUnitParser.java:83)
    at hudson.tasks.junit.JUnitResultArchiver.parse(JUnitResultArchiver.java:123)
    at hudson.tasks.junit.JUnitResultArchiver.perform(JUnitResultArchiver.java:135)
    at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:19)
    at hudson.model.AbstractBuild$AbstractRunner.perform(AbstractBuild.java:649)
    at hudson.model.AbstractBuild$AbstractRunner.performAllBuildSteps(AbstractBuild.java:625)
    at hudson.model.AbstractBuild$AbstractRunner.performAllBuildSteps(AbstractBuild.java:603)
    at hudson.model.Build$RunnerImpl.post2(Build.java:161)
    at hudson.model.AbstractBuild$AbstractRunner.post(AbstractBuild.java:572)
    at hudson.model.Run.run(Run.java:1386)
    at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:46)
    at hudson.model.ResourceController.execute(ResourceController.java:88)
    at hudson.model.Executor.run(Executor.java:145)
Finished: FAILURE
```

Here is an exampe of a valid junit output as generated by [nose](http://somethingaboutorange.com/mrl/projects/nose/):

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuite name="nosetests" tests="175" errors="0" failures="0" skip="0">
  <testcase classname="pyrad.tests.testBidict.BiDictTests" name="testBackwardAccess" time="0"/>
  <testcase classname="pyrad.tests.testBidict.BiDictTests" name="testBackwardDeletion" time="0"/>
  <testcase classname="pyrad.tests.testBidict.BiDictTests" name="testDeletion" time="0"/>
  <testcase classname="pyrad.tests.testBidict.BiDictTests" name="testForwardAccess" time="0"/>
  <testcase classname="pyrad.tests.testBidict.BiDictTests" name="testItemAccessor" time="0"/>
  <testcase classname="pyrad.tests.testBidict.BiDictTests" name="testLength" time="0"/>
  <testcase classname="pyrad.tests.testBidict.BiDictTests" name="testStartEmpty" time="0"/>
  <testcase classname="pyrad.tests.testClient.ConstructionTests" name="testNamedParameters" time="0"/>
  <testcase classname="pyrad.tests.testClient.ConstructionTests" name="testParameterOrder" time="0"/>
  <testcase classname="pyrad.tests.testClient.ConstructionTests" name="testSimpleConstruction" time="0"/>
  <testcase classname="pyrad.tests.testClient.OtherTests" name="testCreateAcctPacket" time="0"/>
  <testcase classname="pyrad.tests.testClient.OtherTests" name="testCreateAuthPacket" time="0"/>
  <testcase classname="pyrad.tests.testClient.SocketTests" name="testAuthDelay" time="2"/>
  <testcase classname="pyrad.tests.testClient.SocketTests" name="testBind" time="0"/>
  <testcase classname="pyrad.tests.testClient.SocketTests" name="testBindClosesSocket" time="0"/>
  <testcase classname="pyrad.tests.testClient.SocketTests" name="testDoubleAccountDelay" time="3"/>
  <testcase classname="pyrad.tests.testClient.SocketTests" name="testDoubleRetry" time="0"/>
  <testcase classname="pyrad.tests.testClient.SocketTests" name="testIgnorePacketError" time="1"/>
  <testcase classname="pyrad.tests.testClient.SocketTests" name="testInvalidReply" time="1"/>
  <testcase classname="pyrad.tests.testClient.SocketTests" name="testNoRetries" time="0"/>
  <testcase classname="pyrad.tests.testClient.SocketTests" name="testReopen" time="0"/>
  <testcase classname="pyrad.tests.testClient.SocketTests" name="testSendPacket" time="0"/>
  <testcase classname="pyrad.tests.testClient.SocketTests" name="testSingleAccountDelay" time="2"/>
  <testcase classname="pyrad.tests.testClient.SocketTests" name="testSingleRetry" time="0"/>
  <testcase classname="pyrad.tests.testClient.SocketTests" name="testValidReply" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.AttributeTests" name="testConstructionParameters" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.AttributeTests" name="testInvalidDataType" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.AttributeTests" name="testNamedConstructionParameters" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.AttributeTests" name="testValues" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryInterfaceTests" name="testContainment" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryInterfaceTests" name="testEmptyDictionary" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryInterfaceTests" name="testReadonlyContainer" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testAttributeEncryptionError" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testAttributeOptions" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testAttributeTooFewColumnsError" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testAttributeUnknownTypeError" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testAttributeUnknownVendorError" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testBeginVendorParsing" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testBeginVendorTooFewColumns" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testBeginVendorUnknownVendor" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testDictFileParseError" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testDictFilePostParse" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testEndVendorParsing" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testEndVendorUnbalanced" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testEndVendorUnknownVendor" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testInclude" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testIntegerValueParsing" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testParseEmptyDictionary" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testParseMultipleDictionaries" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testParseSimpleDictionary" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testStringValueParsing" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testValueForUnknownAttributeError" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testValueTooFewColumnsError" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testVenderTooFewColumnsError" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testVendorFormatError" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testVendorFormatSyntaxError" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testVendorOptionError" time="0"/>
  <testcase classname="pyrad.tests.testDictionary.DictionaryParsingTests" name="testVendorParsing" time="0"/>
  <testcase classname="pyrad.tests.testHost.ConstructionTests" name="testNamedParameters" time="0"/>
  <testcase classname="pyrad.tests.testHost.ConstructionTests" name="testParameterOrder" time="0"/>
  <testcase classname="pyrad.tests.testHost.ConstructionTests" name="testSimpleConstruction" time="0"/>
  <testcase classname="pyrad.tests.testHost.PacketCreationTests" name="testCreateAcctPacket" time="0"/>
  <testcase classname="pyrad.tests.testHost.PacketCreationTests" name="testCreateAuthPacket" time="0"/>
  <testcase classname="pyrad.tests.testHost.PacketCreationTests" name="testCreatePacket" time="0"/>
  <testcase classname="pyrad.tests.testHost.PacketSendTest" name="testSendPacket" time="0"/>
  <testcase classname="pyrad.tests.testHost.PacketSendTest" name="testSendReplyPacket" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AcctPacketConstructionTests" name="testBasicConstructor" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AcctPacketConstructionTests" name="testConstructWithDictionary" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AcctPacketConstructionTests" name="testConstructorDefaults" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AcctPacketConstructionTests" name="testConstructorIgnoredParameters" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AcctPacketConstructionTests" name="testConstructorRawPacket" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AcctPacketConstructionTests" name="testConstructorWithAttributes" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AcctPacketConstructionTests" name="testNamedConstructor" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AcctPacketTests" name="testCreateReply" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AcctPacketTests" name="testRequestPacket" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AcctPacketTests" name="testRequestPacketSetsId" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AcctPacketTests" name="testVerifyAcctRequest" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AuthPacketConstructionTests" name="testBasicConstructor" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AuthPacketConstructionTests" name="testConstructWithDictionary" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AuthPacketConstructionTests" name="testConstructorDefaults" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AuthPacketConstructionTests" name="testConstructorIgnoredParameters" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AuthPacketConstructionTests" name="testConstructorWithAttributes" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AuthPacketConstructionTests" name="testNamedConstructor" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AuthPacketTests" name="testCreateReply" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AuthPacketTests" name="testPwCryptEmptyPassword" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AuthPacketTests" name="testPwCryptPassword" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AuthPacketTests" name="testPwCryptSetsAuthenticator" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AuthPacketTests" name="testPwDecryptEmptyPassword" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AuthPacketTests" name="testPwDecryptPassword" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AuthPacketTests" name="testRequestPacket" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AuthPacketTests" name="testRequestPacketCreatesAuthenticator" time="0"/>
  <testcase classname="pyrad.tests.testPacket.AuthPacketTests" name="testRequestPacketCreatesID" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketConstructionTests" name="testBasicConstructor" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketConstructionTests" name="testConstructWithDictionary" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketConstructionTests" name="testConstructorIgnoredParameters" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketConstructionTests" name="testConstructorWithAttributes" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketConstructionTests" name="testNamedConstructor" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testAddAttribute" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testAttributeAccess" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testAttributeValueAccess" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testCreateAuthenticator" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testCreateReply" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testDecodePacketWithAttribute" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testDecodePacketWithBadAttribute" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testDecodePacketWithEmptyAttribute" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testDecodePacketWithEmptyPacket" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testDecodePacketWithInvalidLength" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testDecodePacketWithMultiValuedAttribute" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testDecodePacketWithPartialAttributes" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testDecodePacketWithTooBigPacket" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testDecodePacketWithTwoAttributes" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testDecodePacketWithVendorAttribute" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testDecodePacketWithoutAttributes" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testDelItem" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testEncodeKey" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testEncodeKeyValues" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testGenerateID" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testHasKey" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testHasKeyWithUnknownKey" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testKeys" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testPktDecodeVendorAttribute" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testPktEncodeAttribute" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testPktEncodeAttributes" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testRawAttributeAccess" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testReplyPacket" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testVendorAttributeAccess" time="0"/>
  <testcase classname="pyrad.tests.testPacket.PacketTests" name="testVerifyReply" time="0"/>
  <testcase classname="pyrad.tests.testPacket.UtilityTests" name="testGenerateID" time="0"/>
  <testcase classname="pyrad.tests.testProxy.OtherTests" name="testProcessInput" time="0"/>
  <testcase classname="pyrad.tests.testProxy.OtherTests" name="testProcessInputNonProxyPort" time="0"/>
  <testcase classname="pyrad.tests.testProxy.ProxyPacketHandlingTests" name="testHHandleProxyPacketHandlesWrongPacket" time="0"/>
  <testcase classname="pyrad.tests.testProxy.ProxyPacketHandlingTests" name="testHandleProxyPacketSetsSecret" time="0"/>
  <testcase classname="pyrad.tests.testProxy.ProxyPacketHandlingTests" name="testHandleProxyPacketUnknownHost" time="0"/>
  <testcase classname="pyrad.tests.testProxy.SocketTests" name="testProxyFd" time="0"/>
  <testcase classname="pyrad.tests.testServer.AcctPacketHandlingTests" name="testHandleAcctPacket" time="0"/>
  <testcase classname="pyrad.tests.testServer.AcctPacketHandlingTests" name="testHandleAcctPacketUnknownHost" time="0"/>
  <testcase classname="pyrad.tests.testServer.AcctPacketHandlingTests" name="testHandleAcctPacketWrongPort" time="0"/>
  <testcase classname="pyrad.tests.testServer.AuthPacketHandlingTests" name="testHandleAuthPacket" time="0"/>
  <testcase classname="pyrad.tests.testServer.AuthPacketHandlingTests" name="testHandleAuthPacketUnknownHost" time="0"/>
  <testcase classname="pyrad.tests.testServer.AuthPacketHandlingTests" name="testHandleAuthPacketWrongPort" time="0"/>
  <testcase classname="pyrad.tests.testServer.OtherTests" name="testAcctProcessInput" time="0"/>
  <testcase classname="pyrad.tests.testServer.OtherTests" name="testAuthProcessInput" time="0"/>
  <testcase classname="pyrad.tests.testServer.OtherTests" name="testCreateReplyPacket" time="0"/>
  <testcase classname="pyrad.tests.testServer.RemoteHostTests" name="testNamedConstruction" time="0"/>
  <testcase classname="pyrad.tests.testServer.RemoteHostTests" name="testSimpleConstruction" time="0"/>
  <testcase classname="pyrad.tests.testServer.ServerConstructiontests" name="testBindDuringConstruction" time="0"/>
  <testcase classname="pyrad.tests.testServer.ServerConstructiontests" name="testParameterOrder" time="0"/>
  <testcase classname="pyrad.tests.testServer.ServerConstructiontests" name="testSimpleConstruction" time="0"/>
  <testcase classname="pyrad.tests.testServer.ServerRunTests" name="testRunIgnoresPacketErrors" time="0"/>
  <testcase classname="pyrad.tests.testServer.ServerRunTests" name="testRunIgnoresPollErrors" time="0"/>
  <testcase classname="pyrad.tests.testServer.ServerRunTests" name="testRunIgnoresServerPacketErrors" time="0"/>
  <testcase classname="pyrad.tests.testServer.ServerRunTests" name="testRunInitializes" time="0"/>
  <testcase classname="pyrad.tests.testServer.ServerRunTests" name="testRunRunsProcessInput" time="0"/>
  <testcase classname="pyrad.tests.testServer.SocketTests" name="testBind" time="0"/>
  <testcase classname="pyrad.tests.testServer.SocketTests" name="testGrabPacket" time="0"/>
  <testcase classname="pyrad.tests.testServer.SocketTests" name="testPrepareSocketAcctFds" time="0"/>
  <testcase classname="pyrad.tests.testServer.SocketTests" name="testPrepareSocketAuthFds" time="0"/>
  <testcase classname="pyrad.tests.testServer.SocketTests" name="testPrepareSocketNoFds" time="0"/>
  <testcase classname="pyrad.tests.testTools.EncodingTests" name="testAddressDecoding" time="0"/>
  <testcase classname="pyrad.tests.testTools.EncodingTests" name="testAddressEncoding" time="0"/>
  <testcase classname="pyrad.tests.testTools.EncodingTests" name="testDateDecoding" time="0"/>
  <testcase classname="pyrad.tests.testTools.EncodingTests" name="testDateEncoding" time="0"/>
  <testcase classname="pyrad.tests.testTools.EncodingTests" name="testDecodeFunction" time="0"/>
  <testcase classname="pyrad.tests.testTools.EncodingTests" name="testEncodeFunction" time="0"/>
  <testcase classname="pyrad.tests.testTools.EncodingTests" name="testIntegerDecoding" time="0"/>
  <testcase classname="pyrad.tests.testTools.EncodingTests" name="testIntegerEncoding" time="0"/>
  <testcase classname="pyrad.tests.testTools.EncodingTests" name="testInvalidAddressEncodingRaisesTypeError" time="0"/>
  <testcase classname="pyrad.tests.testTools.EncodingTests" name="testInvalidDataEncodingRaisesTypeError" time="0"/>
  <testcase classname="pyrad.tests.testTools.EncodingTests" name="testInvalidIntegerEncodingRaisesTypeError" time="0"/>
  <testcase classname="pyrad.tests.testTools.EncodingTests" name="testInvalidStringEncodingRaisesTypeError" time="0"/>
  <testcase classname="pyrad.tests.testTools.EncodingTests" name="testStringDecoding" time="0"/>
  <testcase classname="pyrad.tests.testTools.EncodingTests" name="testStringEncoding" time="0"/>
  <testcase classname="pyrad.tests.testTools.EncodingTests" name="testUnknownTypeDecoding" time="0"/>
  <testcase classname="pyrad.tests.testTools.EncodingTests" name="testUnknownTypeEncoding" time="0"/>
  <testcase classname="pyrad.tests.testTools.EncodingTests" name="testUnsignedIntegerEncoding" time="0"/>
</testsuite>
```

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/wichert"><img src="https://avatars.githubusercontent.com/u/199657?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [wichert](https://github.com/wichert)**
_Sunday Apr 24, 2011 at 19:34 GMT_

----

[This stackoverflow discussion](http://stackoverflow.com/questions/442556/spec-for-junit-xml-output) might have useful information.

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/philsquared"><img src="https://avatars.githubusercontent.com/u/92878?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [philsquared](https://github.com/philsquared)**
_Tuesday Apr 26, 2011 at 09:57 GMT_

----

Thanks @wichert. I'm going to have to have another look at this.
I did see that stackoverflow question (and all that it links to) when I was originally looking into it.
I think the only way I'm going to get this working is to install an instance of Jenkins and/ or Hudson myself

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/wichert"><img src="https://avatars.githubusercontent.com/u/199657?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [wichert](https://github.com/wichert)**
_Tuesday Apr 26, 2011 at 13:08 GMT_

----

If you need another example: I put some [zope testrunner output](http://www.wiggy.net/tmp/report.tar.gz) online as well. One thing that testrunner is does create a separate file for each source file containing tests so you end up with lots of XML files.

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/wichert"><img src="https://avatars.githubusercontent.com/u/199657?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [wichert](https://github.com/wichert)**
_Wednesday Sep 21, 2011 at 14:16 GMT_

----

Can you provide a status update for this ticket?

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/philsquared"><img src="https://avatars.githubusercontent.com/u/92878?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [philsquared](https://github.com/philsquared)**
_Wednesday Sep 21, 2011 at 14:33 GMT_

----

I'm really sorry, Wichert. I managed to drop this somewhere along the line.
I'm pretty snowed at the moment so I don't know when I will be able to get back to it.
Will do my best.
If you fancy looking into it yourself you'd be very welcome, of course :-)

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/wichert"><img src="https://avatars.githubusercontent.com/u/199657?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [wichert](https://github.com/wichert)**
_Thursday Sep 22, 2011 at 15:55 GMT_

----

I'll give it a try. I was still using an older version of Catch, so I'll have to test upgrading first. The first thing I see is lots of new warnings produced by catch and a compile error. so it doesn't appear to be a trivial upgrade.

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/wichert"><img src="https://avatars.githubusercontent.com/u/199657?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [wichert](https://github.com/wichert)**
_Friday Sep 23, 2011 at 09:38 GMT_

----

I think we need to think a bit about how to structure the output. As an example lets assume tests that are setup like this:

``` cpp
TEST_CASE("ClassA", "Short description of class A") {
    SECTION("methodOne", "Tests for method one") {
        SECTION("situation-1", "What happens if XYZ") {
            REQUIRE(...);
            REQUIRE(...);
            REQUIRE(...);
        }
         SECTION("situation-2", "What happens if XYZ") {
            REQUIRE(...);
            REQUIRE(...);
            REQUIRE(...);
        }
    }

    SECTION("methodTwo", "Tests for method two") {
        SECTION("situation-1", "What happens if XYZ") {
            REQUIRE(...);
            REQUIRE(...);
            REQUIRE(...);
        }
         SECTION("situation-2", "What happens if XYZ") {
            REQUIRE(...);
            REQUIRE(...);
            REQUIRE(...);
        }
    }
}

TEST_CASE("ClassB", "Short description of class B") {
    SECTION("methodOne", "Tests for method one") {
        // Repeat similar structure as for classA
```

when reporting results for these tests in JUnit format we run into one problem: Catch supports arbitrary nesting of sections, while JUnit only supports two levels (testsuite -> testcase). I suggest that the simplest thing to do here is to only take the two top levels of Catch (TEST_CASE and top level SECTION). That results in this JUnit structure:

``` xml
<testsuites>
  <testsuite name="ClassA">
    <testcase name="Tests for method one" classname="methodOne" />
    <testcase name="Tests for method two" classname="methodTwo" />
   </testsuite>
  <testsuite name="ClassB">
    <testcase name="Tests for method one" classname="methodOne" />
    <testcase name="Tests for method two" classname="methodTwo" />
   </testsuite>
</testsuites>
```

(ignoring the mandatory attributes such as time, tests, failures, etc.). Looking at the data passed to a Catch reporter this does not match correctly: the top level there is a group, but I never see more than one group being generated. I'm not sure if that is due to me not knowing how to make groups, if that is an not fully implemented feature in Catch or a design flaw (unused group level). 

Can you provide some guidance how I should proceed with this?

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/wichert"><img src="https://avatars.githubusercontent.com/u/199657?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [wichert](https://github.com/wichert)**
_Thursday Nov 10, 2011 at 21:53 GMT_

----

Hi Phil. Can you spare a few minutes to give me a few tips so I can try to fix this?

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/wichert"><img src="https://avatars.githubusercontent.com/u/199657?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [wichert](https://github.com/wichert)**
_Tuesday Mar 13, 2012 at 09:05 GMT_

----

Hi Phil. Is this still on your radar?

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/philsquared"><img src="https://avatars.githubusercontent.com/u/92878?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [philsquared](https://github.com/philsquared)**
_Tuesday Mar 13, 2012 at 17:19 GMT_

----

Sorry Wichert, I've still not really caught up.
Hoping to do a big push soon…

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/wichert"><img src="https://avatars.githubusercontent.com/u/199657?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [wichert](https://github.com/wichert)**
_Tuesday Aug 28, 2012 at 08:45 GMT_

----

Hi Phil. I'm still willing to poke at this, but need some input from you. Is this issue still on your radar?

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/philsquared"><img src="https://avatars.githubusercontent.com/u/92878?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [philsquared](https://github.com/philsquared)**
_Wednesday Aug 29, 2012 at 07:38 GMT_

----

Hey Wichert,

Thanks for your patience. I'm really sorry I've not been getting back to this - and thanks for your help.
This has not dropped off my radar - but I'm trying to get through some work that will impact the reporter interface before I do too much with the JUnit reporter. I'm aiming to get that done in the next week or so.

As for your the question you raised before (sorry I didn't respond before - that was inbox syndrome):

The way I have been approaching it is that each isolated test case run (one for each leaf section) is a JUnit "test". The name can be constructed from the test case name + section name(s) (I need to formalise that naming scheme too).

JUnit test _suites_ are then each set of tests that are matched by the filters provided by a single -t switch on the command line (if no -t is provided then there is just a single suite). Bear in mind that -t can have multiple filters. So, e.g:

-t abc/def\* ghi/jkl\* -t random/1 random/2

Gives us two suites - one named, "abc/def\* ghi/jkl*", and one named, "random/1 random/2".
I have support for naming suites coming too - and this is all being captured in the TestCaseFilters class and will be passed on to the reporter (it's not, yet).

Does that make sense?

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/wichert"><img src="https://avatars.githubusercontent.com/u/199657?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [wichert](https://github.com/wichert)**
_Thursday Aug 30, 2012 at 08:48 GMT_

----

This matches the naming scheming, which I admint I have never understood in catch. I am always struggling with two things:
- why should I have to provide a description for test cases and sections? I already use descriptive names, so the description is often the exact same string. Having to provide it is just annoying.
- how do test case names map to the commandline option for the testrunner? I would expect that if I have a test case named "foo" with a nested testcase named "bar", which contains a section named "buz" I can use options like -t foo, -t foo/bar or -t foo/bar/buz . That never worked, at which point I always gave up trying to use the -t option.

That aside there is a problem with junit XML support: Catch supports nested test cases which does not map to the junit scheme which only supports three levels: package, class and test. Trying force Catch structure into that I would expect something like this:
- each top level test case is a package
- each second level test case is a class. If no second level is used insert a dummy level?
- each section is a test

or alternatively:
- each source file is a package
- each top level test case is a class
- each section is a test

The latter matches python (and I'm guessing java) better, but I am not sure if you can get the source filename in a useful way.

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/philsquared"><img src="https://avatars.githubusercontent.com/u/92878?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [philsquared](https://github.com/philsquared)**
_Friday Aug 31, 2012 at 21:15 GMT_

----

The idea is that the test name is a short, hierarchical, name - something like: "stuff/sub stuff/details" - so all "stuff" tests can be grouped, and all "stuff/sub stuff" tests can be grouped.
You are not forced to work that way (and I don't always) but it's a useful convention.
If you do that then its nice to be able to supply a more detailed description string alongside it too. But I have to admit I use the description string a lot less than I expected to - a lot of the time I leave it to "" (you don't have to put anything the string - but, due to C++98 not having variadic macros you must provide at least the empty string).

Same with sections, although the hierarchical part is less common as they are already in a hierarchy - so I use descriptions even less there.

If you have used hierarchical test case naming (ignore sections for a moment) then you might have tests like:

"a/b/c"
"a/b/d"
"a/e/f"
"g"

Now you can run the first three with `"a*"`, the first two with `"a/b*"` or just the first one with `"a/b/c"`

The idea has always been that this ability would seamlessly extend to sections too - so a section, `"1"` in the first test case, could be run with `"a/b/c/1"`.
But I have not yet implemented that - and there are some subtleties to it that make me wonder if it would work that way.

Nonetheless, the ability to create groups on-the-fly using wildcards (especially with hierarchical naming) has been useful - and has got even more useful just recently as I have added prefix wildcards (`*foo*`) and exclusions (`exclude:foo - or ~foo`) - which can also be mixed (`~*foo*`). Additionally you can supply a series of these filters and the group will be the union of the inclusions, less the union of the exclusions.

So a command line like:

`-t foo/bar* -t a* ~*b*`

Would run two groups. The first is `"foo/bar*"` and will match anything that starts with foo/bar.
The second is `"a* ~*b*"` and will match anything that starts with a, except anything that contains b.

Two features that are coming: named groups and tags (non hierarchical matching) should make this approach more powerful.

So my scheme was to map groups to Junit suites.

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/wichert"><img src="https://avatars.githubusercontent.com/u/199657?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [wichert](https://github.com/wichert)**
_Saturday Sep 01, 2012 at 06:38 GMT_

----

What I am struggling with is the need to prefix fhings manually. To 
illustrate this is how I usually structure my tests:

``` C++
TEST_CASE("MyClass") {
     TEST_CASE("SomeMethod") {
         SECTION("situation-one") {
             ....
            }
         SECTION("situation-two") {
             ....
            }
     }
}
```

I would expect to be able to tell Catch to run tests for MyClass using 
"-t MyClass", or just the tests for MyClass::SomeMethod using "-t 
MyClass/SomeMethod", or even a specific situation using "-t 
MyClass/SomeMethod/situation-one". I do not see why I should be forced 
to manually repeat the hierarchy structure in the test case name as 
Catch is currently forcing me to do:

``` C++
TEST_CASE("MyClass") {
     TEST_CASE("MyClas/SomeMethod") {
         SECTION("MyClass/SomeMethod/situation-one") {
             ....
            }
         SECTION("MyClass/SomeMethod/situation-two") {
             ....
            }
     }
}
```

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/philsquared"><img src="https://avatars.githubusercontent.com/u/92878?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [philsquared](https://github.com/philsquared)**
_Sunday Sep 02, 2012 at 17:39 GMT_

----

I'm not sure where the outer TEST_CASE is coming from.
TEST_CASEs can appear at only one level (they are implemented as free functions).
SECTIONS may be arbitrarily nested within TEST_CASEs (they are implemented as if statements with scoped objects).

It is true the test cases that logically belong together require a common prefix for the hierarchical matching to work. But it's precisely that prefixing that logically groups them together.

Sections do not require prefixes, however. So I believe your example would be written something like:

```
TEST_CASE("MyClass/SomeMethod", "") {
    SECTION("situation-one", "") {
        ....
    }
    SECTION("situation-two", "") {
        ....
    }
}
```

Which is not so bad.

Now you can run all MyClass tests with:

`-t MyClass*`

And everything in that first test case with:

`-t MyClass/SomeMethod`

Unfortunately, at time of writing, you cannot selectively run Sections within a test case. There are some issues around this that make it not straightforward, but I believe I should be able to get it working to some approximation (the main issue is that discovery of sections only occurs as the test case is running).

Does that clarify anything at all? Is it the section selection that you are particularly missing?

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/wichert"><img src="https://avatars.githubusercontent.com/u/199657?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [wichert](https://github.com/wichert)**
_Monday Sep 03, 2012 at 08:15 GMT_

----

I misremembered by code - I was nesting SECTIONs instead of TEST_CASEs. That makes my code look like this:

``` C++
TEST_CASE("MyClass") {
     SECTION("SomeMethod") {
         SECTION("situation-one") {
             ....
         }
         SECTION("situation-two") {
             ....
         }
     }
}
```

 I still do not see why I need to repeat a prefix in the section name when I already create an explicit nesting level in the code. Is there no way to avoid that?

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/wichert"><img src="https://avatars.githubusercontent.com/u/199657?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [wichert](https://github.com/wichert)**
_Monday Sep 03, 2012 at 08:16 GMT_

----

Section selection is certainly something that I am missing. Especially when debugging I often want to run a single (leaf) section so I can test for unintended side-effects and side breakpoints easily without having to worry about other tests triggering them.

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/TypicalFooBar"><img src="https://avatars.githubusercontent.com/u/2712414?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [TypicalFooBar](https://github.com/TypicalFooBar)**
_Saturday Nov 03, 2012 at 02:13 GMT_

----

Hey, I just wanted to chime in here on this conversation of two years :)

I experienced the same error in Jenkins that you were experiencing Wichert.  The NullPointerException because of the output format of the JUnit reporter.

I was able to figure out why Jenkins is crashing, and I have an ugly workaround that lets Jenkins continue to run, but it doesn't solve this issue.

Currently, in the JunitReporter class, there is a function that looks like this:

```
void OutputTestCases( XmlWriter& xml, const Stats& stats ) {
            std::vector<TestCaseStats>::const_iterator it = stats.m_testCaseStats.begin();
            std::vector<TestCaseStats>::const_iterator itEnd = stats.m_testCaseStats.end();
            for(; it != itEnd; ++it ) {
                xml.writeBlankLine();
                xml.writeComment( "Test case" );

                XmlWriter::ScopedElement e = xml.scopedElement( "testcase" );
                xml.writeAttribute( "classname", it->m_className );
                xml.writeAttribute( "name", it->m_name );
                xml.writeAttribute( "time", "tbd" );

                OutputTestResult( xml, *it );
            }
```

Using the above, the output from the JunitReporter looks like the following:

```
<testsuites>
  <testsuite errors="0" failures="0" tests="1" hostname="tbd" time="tbd" timestamp="tbd">

    <!--Test case-->
    <testcase name="SimpleTest" time="tbd"/>
  </testsuite>
  <system-out/>
  <system-err/>
</testsuites>
```

After some trial and error, I found that what was making Jenkins crash was the fact that the testcase element did NOT have the classname attribute.  This is the line of code in the OutputTestCases() function of the JunitReporter class that causes the trouble:

```
xml.writeAttribute( "classname", it->m_className );
```

In my case, this line was not actually printing the classname.  Perhaps it->m_className is null?  When I switched out that line for a random string of "foo", it printed.  The modified OutputTestCases() function looks like this:

```
void OutputTestCases( XmlWriter& xml, const Stats& stats ) {
            std::vector<TestCaseStats>::const_iterator it = stats.m_testCaseStats.begin();
            std::vector<TestCaseStats>::const_iterator itEnd = stats.m_testCaseStats.end();
            for(; it != itEnd; ++it ) {
                xml.writeBlankLine();
                xml.writeComment( "Test case" );

                XmlWriter::ScopedElement e = xml.scopedElement( "testcase" );
                xml.writeAttribute( "classname", "foo" );
                xml.writeAttribute( "name", it->m_name );
                xml.writeAttribute( "time", "tbd" );

                OutputTestResult( xml, *it );
            }
        }
```

With the above change the JunitReporter output looks like this:

```
<testsuites>
  <testsuite errors="0" failures="0" tests="1" hostname="tbd" time="tbd" timestamp="tbd">

    <!--Test case-->
    <testcase classname="foo" name="SimpleTest" time="tbd"/>
  </testsuite>
  <system-out/>
  <system-err/>
</testsuites>
```

Jenkins seems to be okay with this, though it displays the SimpleTest result as if it were a part of the class foo, which is the ugly part I was talking about (but at least it didn't crash!).

I hope this helps you debug the JunitReporter.  Great work on Catch Phil!  It is VERY easy to use, which I really appreciate!  :)

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/philsquared"><img src="https://avatars.githubusercontent.com/u/92878?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [philsquared](https://github.com/philsquared)**
_Monday Nov 05, 2012 at 08:18 GMT_

----

I hadn't realised the missing class name attribute was the only thing (or, at least, the key thing) stopping this from working! (I've still not had a chance to get a Jenkins/ Hudson set-up going to try it for myself).

I've plumbed that attribute in now (it was always using an empty string before).
For test cases that are actually based on methods it will use the class name. For free standing test cases it just uses the string, "global". I'm open to suggestions on making that more useful. But, from the sounds of it, that should unblock you from needing your workaround and at least be a smoother experience.

I realise there are some other attributes that are still set to "tbd". I don't know how important they are to getting this working too.

I've committed those changes to the Interation branch. Would appreciate if you (both) could let me know how that works for you.

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/philsquared"><img src="https://avatars.githubusercontent.com/u/92878?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [philsquared](https://github.com/philsquared)**
_Monday Nov 05, 2012 at 08:20 GMT_

----

@wichert I noticed, while replying above, that our previous discussion never fully concluded. But I was bit confused about this:

"I still do not see why I need to repeat a prefix in the section name when I already create an explicit nesting level in the code. Is there no way to avoid that?"

I don't see any repeated prefix in the last code you posted.

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/wichert"><img src="https://avatars.githubusercontent.com/u/199657?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [wichert](https://github.com/wichert)**
_Monday Nov 05, 2012 at 08:24 GMT_

----

@philsquared are you mixing up this discussion and another issue where we talked about how -s behaves?

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/philsquared"><img src="https://avatars.githubusercontent.com/u/92878?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [philsquared](https://github.com/philsquared)**
_Monday Nov 05, 2012 at 08:28 GMT_

----

No, it was part of this thread:

Hard link here: https://github.com/philsquared/Catch/issues/5#issuecomment-8232083

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/wichert"><img src="https://avatars.githubusercontent.com/u/199657?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [wichert](https://github.com/wichert)**
_Monday Nov 05, 2012 at 08:38 GMT_

----

I have to admit I don't remember exactly what my thoughts were at the time. Some current thoughts:
- there is a prefix involved here: you use MyClass as a prefix (through the `MyClass*` glob.
- being able to run individual sections within a test case is _extremely_ useful
- this should also work if those sections are nested
- perhaps filtering on sections should be a separate commandline option (Python's zope.testrunner uses the -t parameter to do that, nosetests uses --tests). I'm not sure how that will work with nested sections; you may need to define a separating character and advice people not to use that in test names

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/TypicalFooBar"><img src="https://avatars.githubusercontent.com/u/2712414?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [TypicalFooBar](https://github.com/TypicalFooBar)**
_Monday Nov 05, 2012 at 16:42 GMT_

----

Phil,

I just used the _catch.hpp_ file in the Integration branch and Jenkins was quite happy with the output!  I tested it with different successful and unsuccessful test cases, and it seems to be working just fine.

Thank you for taking time to fix this problem!  The other _tbd_ values still need work, but at least Jenkins can read and display the output from the JunitReporter.

Thanks again for your help, and again great job on Catch!

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/philsquared"><img src="https://avatars.githubusercontent.com/u/92878?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [philsquared](https://github.com/philsquared)**
_Monday Nov 05, 2012 at 16:52 GMT_

----

That's great! Thanks for letting me know, Derek.
At this point, since we have at least seen it running now, and due to this thread going on a bit - mostly on peripheral issues - I'm going to close this issue.

@wichert - if it still doesn't work at all for you please reopen.

For any othe JUnit related issues now please raise a new issue.

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/SebDyn"><img src="https://avatars.githubusercontent.com/u/2786959?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [SebDyn](https://github.com/SebDyn)**
_Tuesday Nov 13, 2012 at 13:10 GMT_

----

I want to reopen this bug because it does not work with me.

I use Jenkins ver. 1.479
Jenkins Plugin: xunit ver. 1.50 ( https://wiki.jenkins-ci.org/display/JENKINS/xUnit+Plugin ) 
CATCH: single header distribution. Branch Integration, revision 88b70828f2

Jenkins console output:

```
Gestartet durch Benutzer anonymous
Building in workspace /var/lib/jenkins/jobs/Test Executables/workspace
[SSH] executing pre build script:

/HOMEDIR/vmwarectrl/opensuse/vmstart_test.sh
catched variables: branch=next, cores=4, mode=release, os=linux
[SSH] exit-status: 0
[workspace] $ /bin/sh -xe /tmp/hudson7292242557306021101.sh
[workspace] $ /bin/sh -xe /tmp/hudson4347723387053783514.sh
[SSH] executing post build script:
BUILD_ID="2012-11-13_13-56-01"

ssh rlampert@opensuse /HOMEDIR/devel/build/exec_test.sh -e expensive -e meshdata
mkdir /var/lib/jenkins/jobs/Test\ Executables/workspace/$BUILD_ID
scp rlampert@opensuse:/HOMEDIR/devel/build/*.xml /var/lib/jenkins/jobs/Test\ Executables/workspace/$BUILD_ID
scp rlampert@opensuse:/HOMEDIR/devel/build/test*.txt /var/lib/jenkins/jobs/Test\ Executables/workspace/$BUILD_ID
touch /var/lib/jenkins/jobs/Test\ Executables/workspace/test.xml
[SSH] exit-status: 0
[xUnit] [INFO] - Starting to record.
[xUnit] [INFO] - Processing JUnit
[xUnit] [INFO] - [JUnit] - 1 test report file(s) were found with the pattern '*.xml' relative to '/var/lib/jenkins/jobs/Test Executables/workspace' for the testing framework 'JUnit'.
ERROR: Publisher org.jenkinsci.plugins.xunit.XUnitPublisher aborted due to exception
java.lang.NullPointerException
    at hudson.tasks.junit.CaseResult.getPackageName(CaseResult.java:266)
    at hudson.tasks.junit.TestResult.freeze(TestResult.java:576)
    at hudson.tasks.junit.TestResultAction.setResult(TestResultAction.java:74)
    at hudson.tasks.junit.TestResultAction.<init>(TestResultAction.java:67)
    at org.jenkinsci.plugins.xunit.XUnitPublisher.recordTestResult(XUnitPublisher.java:253)
    at org.jenkinsci.plugins.xunit.XUnitPublisher.performXUnit(XUnitPublisher.java:123)
    at org.jenkinsci.plugins.xunit.XUnitPublisher.perform(XUnitPublisher.java:93)
    at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:19)
    at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:717)
    at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:692)
    at hudson.model.Build$BuildExecution.post2(Build.java:183)
    at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:639)
    at hudson.model.Run.execute(Run.java:1527)
    at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:46)
    at hudson.model.ResourceController.execute(ResourceController.java:88)
    at hudson.model.Executor.run(Executor.java:236)
Sending e-mails to: XXXX
Finished: FAILURE
```

The XML generated by CATCH 

``` xml
<testsuites>
  <testsuite errors="0" failures="0" tests="587" hostname="tbd" time="tbd" timestamp="tbd">

    <!--Test case-->
    <testcase classname="global" name="boost/spirit/width" time="tbd"/>

    <!--Test case-->
    <testcase classname="global" name="sos_import/lsdyna/simple" time="tbd"/>

    <!--Test case-->
    <testcase classname="global" name="sos_import/lsdyna/simple/individual_files" time="tbd"/>

    <!--Test case-->
    <testcase classname="global" name="sos_import/lsdyna/simple/project" time="tbd"/>

    <!--Test case-->
    <testcase classname="global" name="sos_import/lsdyna/parser/old" time="tbd"/>

    <!--Test case-->
    <testcase classname="global" name="sos_import/osl3/bin/simple" time="tbd"/>

    <!--Test case-->
    <testcase classname="global" name="sos_import/osl3/bin/xml" time="tbd"/>

    <!--Test case-->
    <testcase classname="global" name="sos_import/lsdyna/osl3/bsp1_part321" time="tbd"/>

    <!--Test case-->
    <testcase classname="global" name="sos_import/lsdyna/fixed_width" time="tbd"/>

    <!--Test case-->
    <testcase classname="global" name="sos_import/lsdyna/fixed_width/output" time="tbd"/>
  </testsuite>
  <system-out>
    2012-11-13 13:50:36 [   INFO] thread   0 | Node data in database:
  </system-out>
  <system-err>
2012-11-13 13:51:57 [  ERROR] thread   1 |         Did not find data object with (design, quantity) = ('0004', 'not-existing') in data base.
  </system-err>
</testsuites>
```

**Comment by krvajalmiguelangel** [comment]
<a href="https://github.com/SebDyn"><img src="https://avatars.githubusercontent.com/u/2786959?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [SebDyn](https://github.com/SebDyn)**
_Tuesday Nov 13, 2012 at 13:43 GMT_

----

It looks like if <testsuite> needs name="something". If I add this manually, it works.
If I run the test with 'test -t testcasename', a name is provided and JUnit can use it. A name is only not provided if running all tests (without -t).
